src/compiler/parser.ts

```diff
@@ -5732,6 +5732,17 @@ namespace ts {
                 case SyntaxKind.NoSubstitutionTemplateLiteral:  // foo<T> `...`
                 case SyntaxKind.TemplateHead:                   // foo<T> `...${100}...`
                     return true;
+                // Technically strict mode reserved words are valid identifiers in non-strict mode, and can thus
+                // start an expression. However, when they follow a type argument list and are immediately
+                // preceded by a line break, we don't consider them expression starters.
+                // This supports a semicolon-less style for code like the following:
+                //
+                //   let specialFoo = foo<string>
+                //   let value = ...
+                //
+                case SyntaxKind.InterfaceKeyword:
+                case SyntaxKind.LetKeyword:

```

Comment:
You need to cover more than just these cases and have tests for them - for example

```ts
class C {
    static specialFoo = f<string>
    static bar = 123;
}

class D {
    public specialFoo = f<string>
    public bar = 123;
}

class E {
    private specialFoo = f<string>
    private bar = 123;
}

class F {
    protected specialFoo = f<string>
    protected bar = 123;
}
```